File: Solutions/Easy/3461. Check If Digits Are Equal in String After Operations I/Solution.cs
public class Solution {
    public int CountValidSelections(int[] nums) {
        int n = nums.Length;
        int[] prefixSum = new int[n];
        prefixSum[0] = nums[0];
        for(int i = 1; i < n; i++){
            prefixSum[i] += prefixSum[i - 1] + nums[i];
        }
        bool even = prefixSum[n - 1] % 2 == 0;
        int half = 0, result = 0;
        half = prefixSum[n - 1] / 2;
        for(int i = 0; i < n; i++){
            if(even){
                if(prefixSum[i] == half && nums[i] == 0) result += 2;
            }
            else{
                if((prefixSum[i] == half || prefixSum[i] - 1 == half) && nums[i] == 0) result += 1;
            }
        }
        return result;
    }
}

File: Solutions/Easy/3461. Check If Digits Are Equal in String After Operations I/Solution.py
class Solution:
    def countValidSelections(self, nums: List[int]) -> int:
        n = len(nums)
        sum: List[int] = [nums[0]]
        for i in range(1, n):
            sum.append(nums[i] + sum[- 1])
        half = sum[-1] // 2
        even = sum[-1] % 2 == 0
        result = 0
        for i in range(n):
            if(even):
                if(sum[i] == half and nums[i] == 0):
                    result += 2
            else:
                if((sum[i] == half or sum[i] - 1 == half) and nums[i] == 0):
                    result += 1
        return result
        